src/main/default/lwc/descriptionInput/descriptionInput.js

```diff
@@ -65,28 +129,73 @@ export default class DescriptionInput extends LightningElement {
     return this._validity;
   }
 
+  /**
+   * Returns the value of the input.
+   * @returns {string}
+   */
+  @api get value() {
+    return this._value;
+  }
+
+  /**
+   * Shows an error message in the component if there is an error.
+   * @returns {void}
+   */
+  @api validate() {
+    this._validity = !this.required || !!this._value;
+  }
+
   /**
    * Handles the changes in the input.
-   * @return {void}
+   * @returns {void}
    */
   handleChange = (e) => {
     this._validity = !this.required || !!e.target.value;
     this._value = e.target.value;
+    this.debounceUpdateDescriptionState();
   };
 
   /**
-   * Returns the value of the input.
+   * Updates the engine state and dispatch the analytics actions.
+   * @returns {void}
+   */
+  updateDescriptionState() {
+    this.engine.dispatch(
+      this.actions.updateCaseInput({
+        fieldName: this._fieldName,
+        fieldValue: this.replaceTagsWithSpace(this._value)
+      })
+    );
+    this.engine.dispatch(this.actions.logUpdateCaseField(this._fieldName));
+    this.engine.dispatch(this.actions.fetchCaseClassifications());
+    this.engine.dispatch(this.actions.fetchDocumentSuggestions());
+  }
+
+  get currentProgress() {
+    return (
+      (this.getTextContentLength(this._value) * 100) /
+      this.strongDescriptionLength
+    );
+  }
+
+  /**
+   * Returns the length of the rich text after transforming it to normal text.
+   * @param {string} htmlStr
    * @returns {string}
    */
-  @api get value() {
-    return this._value;
+  getTextContentLength(htmlStr) {
+    const div = document.createElement('div');
+    // eslint-disable-next-line @lwc/lwc/no-inner-html
+    div.innerHTML = htmlStr;
+    return div.innerText.replaceAll(' ', '').length;
   }
 
   /**
-   * Shows an error message in the component if there is an error.
-   * @returns {void}
+   * Returns rich text value after replacing the HTML tags with spaces.
+   * @param {string} htmlStr
+   * @returns {string}
    */
-  @api validate() {
-    this._validity = !this.required || !!this._value;
+  replaceTagsWithSpace(htmlStr) {
+    return htmlStr.replace(/(<([^>]+)>)/gi, ' ');
```

Comment:
Is the inner capture group necessary, it doesn't seem to be?

Also, the regex won't replace this string `<>`.

```suggestion
    return htmlStr.replace(/(<[^>]*>)/gi, ' ');
```